charts/virtual-kubelet/templates/deployment.yaml

```diff
@@ -15,18 +15,38 @@ spec:
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         env:
         - name: AZURE_AUTH_LOCATION
-          value: /etc/virtual-kubelet/credentials.json
+          value: /etc/virtual-kubelet/auth/credentials.json
         - name: ACI_RESOURCE_GROUP
           value: {{ .Values.env.aciResourceGroup }}
         - name: ACI_REGION
           value: {{ default "westus" .Values.env.aciRegion }}
+        - name: APISERVER_CERT_LOCATION
+          value: /etc/virtual-kubelet/apiservercert/cert.pem
+        - name: APISERVER_KEY_LOCATION
+          value: /etc/virtual-kubelet/apiserverkey/key.pem
+        - name: VKUBELET_POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
         volumeMounts:
         - name: credentials
-          mountPath: "/etc/virtual-kubelet"
+          mountPath: "/etc/virtual-kubelet/auth"
+          readOnly: true
+        - name: apiservercert
+          mountPath: "/etc/virtual-kubelet/apiservercert"
+          readOnly: true
+        - name: apiserverkey
+          mountPath: "/etc/virtual-kubelet/apiserverkey"
           readOnly: true
         command: ["virtual-kubelet"]
-        args: ["--provider", "azure", "--nodename", {{ default "aci-conn" .Values.env.nodeName | quote }} , "--os", {{ default "Linux" .Values.env.nodeOsType | quote }}, "--taint", {{ default "azure.com/aci" .Values.env.nodeTaint | quote }}]
+        args: ["--provider", "azure", "--namespace", "default", "--nodename", {{ default "virtual-kubelet" .Values.env.nodeName | quote }} , "--os", {{ default "Linux" .Values.env.nodeOsType | quote }}, "--taint", {{ default "azure.com/aci" .Values.env.nodeTaint | quote }}]
       volumes:
       - name: credentials
         secret:
-          secretName: {{ template "fullname" . }}
\ No newline at end of file
+          secretName: {{ template "fullname" . }}
```

Comment:
why do we create 3 volumes for the same secret? they are mounted in 3 pathes with the same content.

Reply:
There are 3 different files we need to mount. `credentials.json`, `cert.pem`, and `key.pem`.

Reply:
they are in the same secret. you just need to mount in one path, and you can find all 3 files in it.

Reply:
done